homeassistant/components/tplink_omada/device_tracker.py

```diff
@@ -0,0 +1,206 @@
+class OmadaDeviceTrackerEntityDescription(EntityDescription):
+    """Entity description for a device tracker derived from a connected wireless client."""
+
+    coordinator_update_func: Callable[
+        [OmadaClientsCoordinator, str], OmadaWirelessClient | None
+    ]
+    ip_address_fn: Callable[[OmadaWirelessClient | None], str | None]
+    mac_address_fn: Callable[[OmadaWirelessClient | None], str | None]
+    hostname_fn: Callable[[OmadaWirelessClient | None], str | None]
+    location_fn: Callable[[OmadaWirelessClient | None], str | None]
+    is_connected_fn: Callable[[OmadaWirelessClient | None], bool]
+    has_entity_name = True
+
+
+DEVICE_TRACKERS = [
+    OmadaDeviceTrackerEntityDescription(
+        key="device_tracker",
+        coordinator_update_func=lambda coordinator, mac: coordinator.data.get(mac),
+        ip_address_fn=lambda client: client.ip if client else None,
+        mac_address_fn=lambda client: client.mac if client else None,
+        hostname_fn=lambda client: client.host_name if client else None,
+        location_fn=lambda client: client.ap_name if client else None,
+        is_connected_fn=lambda client: client.is_active if client else False,
+    )
+]
+
+
+class OmadaBaseTrackerEntity(CoordinatorEntity[OmadaClientsCoordinator]):
+    """Base entity for tracking a client connected to the Omada network."""
+
+    entity_description: OmadaDeviceTrackerEntityDescription
+    _client_details: OmadaWirelessClient | None = None
+
+    def __init__(
+        self,
+        site_id: str,
+        client_id: str,
+        display_name: str,
+        coordinator: OmadaClientsCoordinator,
+        entity_description: OmadaDeviceTrackerEntityDescription,
+    ) -> None:
+        """Initialize the scanner."""
+        super().__init__(coordinator)
+        self.entity_description = entity_description
+        self._site_id = site_id
+        self._client_id = client_id
+        self._attr_name = display_name
+
+    @property
+    def source_type(self) -> SourceType:
+        """Return the source type of the device."""
+        return SourceType.ROUTER
+
+    def _do_update(self) -> None:
+        self._client_details = self.entity_description.coordinator_update_func(
+            self.coordinator, self._client_id
+        )
+
+    async def async_added_to_hass(self) -> None:
+        """When entity is added to hass."""
+        await super().async_added_to_hass()
+        self._do_update()
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._do_update()
+        self.async_write_ha_state()
+
+
+class OmadaClientScannerEntity(OmadaBaseTrackerEntity, ScannerEntity):
+    """Entity for a client connected to the Omada network."""
+
+    @property
+    def ip_address(self) -> str | None:
+        """Return the primary ip address of the device."""
+        return self.entity_description.ip_address_fn(self._client_details)
```

Comment:
We shouldn't use entity descriptions for the device tracker platform since the base device tracker entity doesn't support that yet.